lib/validator.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+const Joi = require('joi');
+
+module.exports = {
+  auth: Joi.object().keys({
+    credentials: Joi.object().required()
+  }),
```

Comment:
I feel like this is the same case as the `request.filter`. If `auth.credentials` isn't present then it's the responsibility of the `filter` function to fail. The plugin doesn't actually depend on it, so it shouldn't fail preemptively.

What do you think?